### ISSUE TITLE
High Level Descriptions and Emergent Laws

### ISSUE THREAD

**Issue by superintgr** [superintgr/native#2]
I am not going to explain problems in terms of local and asymmetric constructs. All my tasks are going to assume a problem and that problem is going to be described in a context-free universal language. More to be discovered in that project but let this be the first description outlining the basic informal attributes that might be seen here.

**Comment by superintgr** [comment]
**Issue Lingering In My Head For Another Year**

- [ ] Where do you put anything that transcends particular instantiating factors?
- [ ] With no factors to consider, how would the information be used and prepared?
- [ ] Computable objects needing a computer, can't we have natural computers instead of general purpose machines?
- [ ] How do you program the natural machines?

Suppose we are programming something and want to deploy the dynamics in something. We would like those processes to be carried out automatically and by some substrate in which its state is defined as naturally occurring processes.

What if we cannot physically rent out the necessary resources for deployment and running such processes as working independently with cycles that doesn't depend of any particular limiting factors?


**Finding ongoing threads of computations**

There are processes that are taking place in nature whose objective existence depends upon only the environment that doesn't have any arbitrary constraints such as those that would be imposed in running a system of processes using a general purpose computing machinery.

- [ ] Can we tap into the knowledge and augment our computing capabilities?
- [ ] Can we define some tasks along side the existing tasks that are happening in nature?
- [ ] Can a single computer program install the required code?
- [ ] Can we check or measure some observables from time to time without instantiating the whole process as simulation in some form of closed loop?


**Testable regularity in nature is the law of physics**

- [ ] Sample from the device for some N frames over T period
- [ ] Normalize the data
- [ ] Parametrize a substrate with the resource
- [ ] Instantiate other subsystems using the substrate as parent constructor


```python
import torch
from torch.nn.functional import normalize
import util

M = 192000
T = 1
C = 2

resource = util.sample_observables(frames=M, duration=T, channels=C, dtype=torch.float32)

resource = normalize(resource) # shape (192000, 2)

class Substrate(object):
    """
    The base constructor.

    It can instantiate subsystems with resource values by parametrizing their modules and keeping a bijective map between their states.

    From one-time sampling of observables the substrates can evolve discretely any number of subsystems and subsequent sampling will globally update all subsystems associated with the parent constructor simultaneously.

    The subsystem can select their preferred number of blocks and perform further construction using their respective nodes to instantiate subsystems to which they remain as the base constructor.

    Each sampling of observables from the device will yield a two column state vector which has the multiple of 192k values as row components.

    Note: The two columns are jointly isolated, meaning that each of the physical channels have a separate core and parallel structure that independent samples the electromagnetic radiation from separate atoms.
    """
```

**Comment by superintgr** [comment]
**Encoding and Knowledge Graphs**

I have a huge amount of unprocessed text in various physical substrates. The goal is to unify them into a reduced form and extract insights from them.

General requirements:

- [ ] find the symbolic links to all the file paths
- [ ] capture the relative positions and structure of the file tree
- [ ] develop an encoding technique for lossless coding of the file contents
- [ ] create embedding model with maximum size less than 1024 per block of memory
- [ ] connect the embedding network to a structured classification network
- [ ] build a classification table based on k selected characteristics where characteristics refer to the some mode of constant type
- [ ] develop a language model and network of language substrates
- [ ] prepare and attach standardized tasks for each named language substrates
- [ ] using automatic gradient tracking mechanism employ methods that can be applied to push-pull policies



**Terminologies**

_symbolic link_ : 
_mode of constant types_ : 
_classification network_ : 
_embedding network_ : 
_language model_ : 
_language substrate_ : 
_standardized task_ : 
_prepare and attach_ : following some problem specific tasks, a set of attributes of some named system gets defined or adjusted and for all such resources with respect to the original problem description a different system receives the modules where they live as limiting/functional cases.
_automatic gradient tracking_ : refers to PyTorch/JAX etc. tools where the gradient computation is done in the background for some set of compatible computation graph being built.
_push-pull policy_ : a policy is a dictionary and the push-pull key will define criteria for which both levels have unit strength.
_unit strength_ : a variable defined under conditional statements whereby two opposing forces (functional) are equilibrated via a combined weight whose multiple leaves the sum total to remain constant.
_conditional statement_ : for two or more possible states of a system, a filter which defines which state should have dominance based on some quantity being equalized.

**Comment by superintgr** [comment]
**Modules, Gates, Logics, Fields**

- There is a field where things are possible.
- Due to gradient, differentiation is possible.
- Comparable states with respect to controls are found.
- Control logic with respect to ground state is defined.
- Logical states could be read in discrete substrates to build programming languages.

(-0) programmable object
(-1) logical states
(-2) logic gates
(-3) gradient signals
(-4) field functional

**Comment by superintgr** [comment]
**Simulated Workflow**

Consider the following state matrix:

```
[0, 1]
[0, 1]
[1, 0]
[0, 1]
[0, 1]
[0, 1]
[1, 0]
[1, 0]
[1, 0]
[0, 1]
```

**Comment by superintgr** [comment]
- There are 8 pairs of binary variables.
- From start to end, the sequence is continuous.
- The pattern of 0's or 1's along row dimension is either 0 or 1 and never two of the same binary state.
- The same number of times the prior pattern is repeated but with opposite polarity.
- Sum of the squares of each binary variable is equal to 1.
- Mean energy of the values along each column dimension is 1 hence [1, 1] for 


**Designing Instruction Sets**

- [ ] prepare a new block of code execution with a configuration file
- [ ] configuration file will wait for the next opportunity to instantiate a sequential block of memory equal to the total length of runtime
- [ ] configuration file must contain the initialization code for it would callback with that specified block running
- [ ] callback will yield the control into the runtime chain
- [ ] runtime is composed of unit steps where fixed number of steps are completed within a block
- [ ] blocks are labeled region within memory subjected to permutation which takes place when connected to the runtime context

**Comment by superintgr** [comment]
```
[0, 1]
[1, 0]
```

_0_ | 1 : 1
_0_ | 0 : 0
_1_ | 1 : 1
_1_ | 0 : 0

**From the perspective of (channel _10_) | _(_channel_ 01)**

[0], [1] | [0], [0] : _condition_
[0], [1] | [0], [1] : _condition_
[1], [1] | [0], [0] : _condition_
[1], [0] | [0], [1] : _condition_

**Comment by superintgr** [comment]
**Little Program Museum**

- [ ] quietly observe what other processes are involved
- [ ] quietly sense how other portions are doing their job
- [ ] for most of the time do that sensing actively


SupposeWreckegr etsarousedb, ut there'snothingin sightto smash.Then Wrecfterwilhl ave to get somehelp-by putting Builder to work, for example.But what if, at somelater time, Wrickerconsidersthetowertobehighenoughtosmash,whileBuilderwantstomakeit taller still?Who could settlethat dispute?


**The Bureaucrats**

bu.reauc'ra.c/n. the administrationof government throughdepartmentsandsubdivisionms anagedby
setsof officialsfollowinean inflexibleroutine.
I-w ged
nbrnt i.lJnabdri Dictionary

- [ ] Which agents choose which others to do what jobs?
- [ ] Who will decide which jobs are done at all?
- [ ] Who decides what efforts to expend?
- [ ] How will conflicts be settled?

_orders_ : those you are made to take and those that you are forced to give

**Memory**

- [ ] `see` requires a memory to be ready before starting the current task while finishing the previous task
- [ ] use data structure that allows agent to continue working on more than one task in a circular manner without running out of resources

**Ideals**

> _note`_ : the minimal criteria for setting up rivalry is to claim that which the antipode must want or when both parties want the same outcome a superior subordinate the whole level uniformly, leaving their subordinates powerless in the process too

- [ ] Society embeds their ideal as long term memory so that through each others memory their ideal propagates


**Circular Causality**

A causesB "lohn wantedto go homebecausheefelt tiredof work." BcausesA "lohnfelttiredofworkbecausheewantedtogohome." 


**Individual**

- [ ] If Mary says something to James, she must remember to whom it was told
- [ ] Mary must not assume every other person. knows it too
- [ ] Without the concept of individual, we could not have any sense of responsibility


**Minimal Agent & Agency Programming**

Standard Programming Levels:

- [ ] memory, disk, device, processor

**Comment by superintgr** [comment]
**Design Guide**

# Naming Standard

`state your problem abstractly` : `show your invented language` -> `demonstrate your invention in effect`

# Allower / Preventor

 `allowr allows certain dynamics to manifest` : `application = Allowr(application)` -> `policy = Preventr(application)`

# Guarded Runtime

```python
Safari = Allowr(safari)
Browser = Preventr(Safari)

execute(Browser)
```

```bash
Guarded Runtime Started!

05:31:01 safari allowed True
05:31:03 browser prevented False
05:31:10 policy generated True

**Browser Policy**
- website ran via safari
-- safari wrapped via browser
--- wrapper used by user
---- internet addresses of type Wikipedia
--- access granted under preventer permission
-- prevention did remain silent
- allowed exploration under default conditions

Default condition include academic freedom and scientific historical interest

**Pages Blacklisted**

---- YouTube all kinds
--- Google all kinds
-- unassigned
- unassigned

```

**Comment by superintgr** [comment]
> **Design Guide**
> 
> # Naming Standard
> 
> `state your problem abstractly` : `show your invented language` -> `demonstrate your invention in effect`
> 
> # Allower / Preventor
> 
>  `allowr allows certain dynamics to manifest` : `application = Allowr(application)` -> `policy = Preventr(application)`
> 
> # Guarded Runtime
> 
> ```python
> Safari = Allowr(safari)
> Browser = Preventr(Safari)
> 
> execute(Browser)
> ```
> 
> ```bash
> Guarded Runtime Started!
> 
> 05:31:01 safari allowed True
> 05:31:03 browser prevented False
> 05:31:10 policy generated True
> 
> **Browser Policy**
> - website ran via safari
> -- safari wrapped via browser
> --- wrapper used by user
> ---- internet addresses of type Wikipedia
> --- access granted under preventer permission
> -- prevention did remain silent
> - allowed exploration under default conditions
> 
> Default condition include academic freedom and scientific historical interest
> 
> **Pages Blacklisted**
> 
> ---- YouTube all kinds
> --- Google all kinds
> -- unassigned
> - unassigned
> 
> ```
> 

# Key Missing Libraries

- Interface for controlling web browsing behavior
- Safari or chrome or other browser wrapper
- Reliable permission grants
- User interface compatible with exploration
